aioauth/oidc/core/grant_type.py

```diff
@@ -0,0 +1,82 @@
+"""
+.. code-block:: python
+
+    from aioauth.oidc.core import grant_type
+
+Different OAuth 2.0 grant types with OpenID Connect extensions.
+
+----
+"""
+from aioauth.grant_type import (
+    AuthorizationCodeGrantType as OAuth2AuthorizationCodeGrantType,
+)
+from aioauth.models import Client
+from aioauth.oidc.core.responses import TokenResponse
+from aioauth.oidc.core.requests import TRequest
+from aioauth.storage import TStorage
+from aioauth.utils import generate_token
+
+
+class AuthorizationCodeGrantType(OAuth2AuthorizationCodeGrantType[TRequest, TStorage]):
+    """
+    The Authorization Code grant type is used by confidential and public
+    clients to exchange an authorization code for an access token. After
+    the user returns to the client via the redirect URL, the application
+    will get the authorization code from the URL and use it to request
+    an access token.
+    It is recommended that all clients use `RFC 7636 <https://tools.ietf.org/html/rfc7636>`_
+    Proof Key for Code Exchange extension with this flow as well to
+    provide better security.
+
+    Note:
+        Note that ``aioauth`` implements RFC 7636 out-of-the-box.
+        See `RFC 6749 section 1.3.1 <https://tools.ietf.org/html/rfc6749#section-1.3.1>`_.
+    """
+
+    async def create_token_response(
```

Comment:
Seems unfortunate to repeat so much of this. I'm open to ideas if you have any.

Reply:
Hey! Sorry for the delay in my response. Give me a bit more time; I will review it over the weekend.

Reply:
I think it's perfectly normal if the code is duplicated. Later, after finishing the implementation of the "draft", we can think about optimizing the code.